### PR TITLE
Add functions and script to combine audio files and their transcripts

### PR DESCRIPTION
**Pull request by cbrochtrup**
This script can combine audio files and their transcript files. It assumes the audio files are identical length.

### CODE REVIEW

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We require contributors to be identified and we couldn't parse GitHub user details of the following contributors: **Colin Brochtrup**.

In order for your PR to be reviewed and merged, please follow the following directions:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails
4. Amend the authors in the commit history; use `git commit --amend` to only change last commit, or one between `git reset --soft` and `git rebase` to checkout changes, rewrite commit history locally and (force) push changes to the downstream branch

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If you still have any questions or just don't know what to do next, please email [help@finos.org](mailto:help@finos.org). /CC @finos-admin

File: asrtoolkit/combine_audio_files.py [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+
+def check_audio_file(audio_file_name):
+    if valid_input_file(audio_file_name, ["mp3", "sph", "wav", "au", "raw"]):
+        return audio_file(audio_file_name)
+    else:
+        LOGGER.error("Invalid audio file {}".format(audio_file_name))
+
+
+def check_transcript_segment(segment):
+    if not hasattr( segment, 'start' ):
```
Comment (AGiantSquid) [inline asrtoolkit/combine_audio_files.py]:
```suggestion
    if not hasattr(segment, 'start'):
```

File: asrtoolkit/combine_audio_files.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        LOGGER.error("Transcript segment doesn't include the start time, segment: {}".format(segment))
+
+
+def combine_transcripts(transcripts, output_file_name):
+    # Get one list of segments
+    out_transcript = reduce(operator.add, transcripts)
+    out_transcript.location = os.path.join(strip_extension(output_file_name) + '.' +
+                                           out_transcript.file_extension)
+    out_transcript.write(out_transcript.location )
```
Comment (AGiantSquid) [inline asrtoolkit/combine_audio_files.py]:
```suggestion
    out_transcript.write(out_transcript.location)
```

File: asrtoolkit/data_structures/audio_file.py [hunk 3]
```diff
@@ -60,6 +60,17 @@ def degrade_audio(source_audio_file, target_audio_file=None):
     os.remove(tmp2)
 
 
+def combine_audio(audio_files, output_file, gain=False):
+    # Combine audio files with possible remormalization to 0dB
+    gain_str = ''
+    if gain:
+        gain_str = 'gain -n 0'
+    print( audio_file )
```
Comment (AGiantSquid) [inline asrtoolkit/data_structures/audio_file.py]:
```suggestion
    print(audio_file)
```

File: asrtoolkit/data_structures/audio_file.py [hunk 4]
```diff
@@ -60,6 +60,17 @@ def degrade_audio(source_audio_file, target_audio_file=None):
...
+    print( ' '.join(audio_files) )
```
Comment (AGiantSquid) [inline asrtoolkit/data_structures/audio_file.py]:
```suggestion
    print(' '.join(audio_files))
```

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We require contributors to be identified and we couldn't parse GitHub user details of the following contributors: **Colin Brochtrup**.

In order for your PR to be reviewed and merged, please follow the following directions:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails
4. Amend the authors in the commit history; use `git commit --amend` to only change last commit, or one between `git reset --soft` and `git rebase` to checkout changes, rewrite commit history locally and (force) push changes to the downstream branch

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If you still have any questions or just don't know what to do next, please email [help@finos.org](mailto:help@finos.org). /CC @finos-admin

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We require contributors to be identified and we couldn't parse GitHub user details of the following contributors: **Colin Brochtrup**.

In order for your PR to be reviewed and merged, please follow the following directions:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails
4. Amend the authors in the commit history; use `git commit --amend` to only change last commit, or one between `git reset --soft` and `git rebase` to checkout changes, rewrite commit history locally and (force) push changes to the downstream branch

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If you still have any questions or just don't know what to do next, please email [help@finos.org](mailto:help@finos.org). /CC @finos-admin

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We require contributors to be identified and we couldn't parse GitHub user details of the following contributors: **Colin Brochtrup**.

In order for your PR to be reviewed and merged, please follow the following directions:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails
4. Amend the authors in the commit history; use `git commit --amend` to only change last commit, or one between `git reset --soft` and `git rebase` to checkout changes, rewrite commit history locally and (force) push changes to the downstream branch

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If you still have any questions or just don't know what to do next, please email [help@finos.org](mailto:help@finos.org). /CC @finos-admin

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We require contributors to be identified and we couldn't parse GitHub user details of the following contributors: **Colin Brochtrup**.

In order for your PR to be reviewed and merged, please follow the following directions:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails
4. Amend the authors in the commit history; use `git commit --amend` to only change last commit, or one between `git reset --soft` and `git rebase` to checkout changes, rewrite commit history locally and (force) push changes to the downstream branch

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If you still have any questions or just don't know what to do next, please email [help@finos.org](mailto:help@finos.org). /CC @finos-admin

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We require contributors to be identified and we couldn't parse GitHub user details of the following contributors: **Colin Brochtrup**.

In order for your PR to be reviewed and merged, please follow the following directions:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails
4. Amend the authors in the commit history; use `git commit --amend` to only change last commit, or one between `git reset --soft` and `git rebase` to checkout changes, rewrite commit history locally and (force) push changes to the downstream branch

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If you still have any questions or just don't know what to do next, please email [help@finos.org](mailto:help@finos.org). /CC @finos-admin

File: asrtoolkit/data_structures/audio_file.py (hunk 3 above)
Comment (cbrochtrup) [reply asrtoolkit/data_structures/audio_file.py]: Remove print statements in a commit.

File: asrtoolkit/data_structures/audio_file.py (hunk 4 above)
Comment (cbrochtrup) [reply asrtoolkit/data_structures/audio_file.py]: I removed this print statement too.

File: asrtoolkit/combine_audio_files.py [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+def check_audio_file(audio_file_name):
+    if valid_input_file(audio_file_name, ["mp3", "sph", "wav", "au", "raw"]):
+        return audio_file(audio_file_name)
+    else:
+        LOGGER.error("Invalid audio file {}".format(audio_file_name))
+
+
+def check_transcript_segment(segment):
+    if not hasattr(segment, 'start'):
+        LOGGER.error("Transcript segment doesn't include the start time, segment: {}".format(segment))
```
Comment (AGiantSquid) [inline asrtoolkit/combine_audio_files.py]: Should this function throw an exception or return False to stop the workflow, or will the script still work if there are no start times?

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We require contributors to be identified and we couldn't parse GitHub user details of the following contributors: **Colin Brochtrup**.

In order for your PR to be reviewed and merged, please follow the following directions:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails
4. Amend the authors in the commit history; use `git commit --amend` to only change last commit, or one between `git reset --soft` and `git rebase` to checkout changes, rewrite commit history locally and (force) push changes to the downstream branch

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If you still have any questions or just don't know what to do next, please email [help@finos.org](mailto:help@finos.org). /CC @finos-admin

File: asrtoolkit/combine_audio_files.py (hunk 5 above)
Comment (cbrochtrup) [reply asrtoolkit/combine_audio_files.py]: You're right, this script should exit on those errors! Thanks.

Fixed in error handling commit.

**Comment by finos-admin** [comment]
Thank you for your pull request and welcome to our community. We require contributors to be identified and we couldn't parse GitHub user details of the following contributors: **Colin Brochtrup**.

In order for your PR to be reviewed and merged, please follow the following directions:
1. check if your git client is configured with an email to sign commits `git config --list | grep email`
2. If not, set it up using `git config --global user.email email@example.com`
3. Make sure that the git commit email is configured in your GitHub account settings, see https://github.com/settings/emails
4. Amend the authors in the commit history; use `git commit --amend` to only change last commit, or one between `git reset --soft` and `git rebase` to checkout changes, rewrite commit history locally and (force) push changes to the downstream branch

Project team: please **do not merge this Pull Request** until Foundation staff have confirmed that a CLA is in place for the new contributor(s) listed above.

If you still have any questions or just don't know what to do next, please email [help@finos.org](mailto:help@finos.org). /CC @finos-admin

**Comment by cbrochtrup** [comment]
> Let's yapf the PR (style is in root of repo)
> and add this to setup.py
> Otherwise +1

PR has been yapf'd

**Pull request closed without merging**